pkg/filesystem/path/windows_parser.go

```diff
@@ -0,0 +1,78 @@
+package path
+
+import (
+	"strings"
+	"unicode"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+type windowsParser struct {
+	absolute    bool
+	driveLetter rune
+	path        string
+}
+
+// NewWindowsParser creates a Parser for Windows paths that can be used in Resolve.
+func NewWindowsParser(path string) (Parser, error) {
+	// Windows paths are generally passed to system calls that
+	// accept C strings. There is no way these can accept null
+	// bytes.
+	driveLetter := rune(0)
+	absolute := false
+
+	if strings.ContainsRune(path, '\x00') {
+		return nil, status.Error(codes.InvalidArgument, "Path contains a null byte")
+	}
+
+	path = strings.Replace(path, "/", "\\", -1)
+	// C:\
+	if len(path) >= 3 && unicode.IsLetter(rune(path[0])) && path[1] == ':' && path[2] == '\\' {
```

Comment:
Instead of using `unicode.IsLetter()`, I'd recommend using `path[0] >= 'A' && path[0] <= 'Z'`. The reason being that Windows does not permit any non-ASCII characters. The conversion you're doing there (`rune(path[0])`) also means you're interpreting the first byte as if it's ISO-8859-1 instead of UTF-8.

Reply:
Furthermore, you likely also want to put some code here that converts lowercase letters to uppercase.